var/spack/repos/builtin/packages/aws-parallelcluster/package.py

```diff
@@ -37,12 +45,14 @@ class AwsParallelcluster(PythonPackage):
     version('2.5.1', sha256='4fd6e14583f8cf81f9e4aa1d6188e3708d3d14e6ae252de0a94caaf58be76303')
     version('2.5.0', sha256='3b0209342ea0d9d8cc95505456103ad87c2d4e35771aa838765918194efd0ad3')
 
+    depends_on('py-jmespath@0.10', when='@3.0:', type=('build', 'run'))
+
     depends_on('python@3.6:', when='@2.11:', type=('build', 'run'))
     depends_on('python@2.7:2.8,3.4:', when='@:2.10', type=('build', 'run'))
 
     depends_on('py-future@0.16.0:0.18.2', when='@:2.10', type=('build', 'run'))
 
-    depends_on('py-ipaddress@1.0.22:', type=('build', 'run'))
+    depends_on('py-ipaddress@1.0.22:', when='@:2.11', type=('build', 'run'))
 
     depends_on('py-configparser@3.5.0:3.8.1', when='^python@:2', type=('build', 'run'))
 
```

Comment:
According to the `setup.py`, shouldn't it be `py-tabulate@0.8.0:0.999`? Also, I see a ton of missing dependencies:
```python
    "marshmallow~=3.10",                                                                 
    "aws-cdk.core~=" + CDK_VERSION,                                                      
    "aws-cdk.aws-batch~=" + CDK_VERSION,                                                 
    "aws_cdk.aws-cloudwatch~=" + CDK_VERSION,                                            
    "aws-cdk.aws-codebuild~=" + CDK_VERSION,                                             
    "aws-cdk.aws-dynamodb~=" + CDK_VERSION,                                              
    "aws-cdk.aws-ec2~=" + CDK_VERSION,                                                   
    "aws-cdk.aws-efs~=" + CDK_VERSION,                                                   
    "aws-cdk.aws-events~=" + CDK_VERSION,                                                
    "aws-cdk.aws-fsx~=" + CDK_VERSION,                                                   
    "aws-cdk.aws-imagebuilder~=" + CDK_VERSION,                                          
    "aws-cdk.aws-iam~=" + CDK_VERSION,                                                   
    "aws_cdk.aws-lambda~=" + CDK_VERSION,                                                
    "aws-cdk.aws-logs~=" + CDK_VERSION,                                                  
    "aws-cdk.aws-route53~=" + CDK_VERSION,                                               
    "aws-cdk.aws-ssm~=" + CDK_VERSION,                                                   
    "aws-cdk.aws-sqs~=" + CDK_VERSION,                                                   
    "aws-cdk.aws-cloudformation~=" + CDK_VERSION,      
    "werkzeug~=2.0",                                                                     
    # pin connexion to 2.7.0 since 2.8.0,2.9.0 have introduced stricter requirements for werkzeug and flask.
    # This does not impact any functionality. Also next connexion release will lift this limitation
    # and we will be able to upgrade.                                                    
    "connexion==2.7.0",                                                                  
    "flask~=2.0",               
```

Reply:
Hello Adam, 
I don't see marshmallow, aws-cdk, flask 2.0, etc are available on Spack. Should I include them even if they are not available on Spack?

Reply:
You'll need to write packages for these dependencies, otherwise you won't be able to install the package.

Reply:
Is this the list of package on Spack https://spack.readthedocs.io/en/latest/package_list.html? I see the python packages have a prefix `py-`. And there is not marshmallow, aws-cdk, flask 2.0. Should I include them even if they are not available on Spack?

Reply:
Yes, that is the list of packages. You can also get this from the command-line by running `spack list`. You need to add packages for mashmallow, aws-cdk, flask, etc. if you want to install aws-parallelcluster.

Reply:
So adding `depends_on` in this file is good, or we have to talk to our dependencies to let them upload their package on Spackk

Reply:
To install aws-parallelcluster from Spack, do all the dependencies have to be installed from Spack, or some of them could be installed ahead by users from pip?

Reply:
Someone (either you or the developers of those packages) needs to add build recipes for your dependencies. Until then, this PR cannot be merged.

Reply:
Thank you! To make sure I understood, we need first add the build recipes for those absent packages and then add `depends_on` for aws-parallelcluster to those dependencies?

Reply:
You can do everything in this PR or you can submit separate PRs for those dependencies and wait until they get merged, either is fine.